### ISSUE TITLE
Domain for validating container contents

### ISSUE THREAD

**Issue by meganwolf0** [defenseunicorns-labs/lula1#559]
### Is your feature request related to a problem? Please describe.
There's the use case where we need to validate individual FIPS modules inside each container - this most likely will need a specific domain to extract this file data. 

I think we can use the `crane export` command to pull down the file data and then will need to figure out how to extract file specific information and additionally what exactly is going to be checked therein.

I think we'd have to bake in the crane binary under the hood to do this, but hopefully would provide the functionality to support this use case, especially for cgr containers that have no shell.

### Describe the solution you'd like

- **Given** I'm looking for container filesystem info
- **When** a supported domain is used
- **Then** file data is returned to the policy provider

### Describe alternatives you've considered
None

### Additional context
None

**Comment by brandtkeller** [comment]
Less this example and more others (more niche) I wonder if we could validate some behaviors through the "External tool Output file + Lula files domain" approach. 

This proposal makes a lot of sense - containers are not going anywhere. But I do wonder if a `files` domain would help us prove out initial iterations of what we might need natively?

**Comment by meganwolf0** [comment]
Yeah I think a `files` domain would provide some of the functionality - I would assume that would just like grab a target (local or remote) and convert it to a map? I was being more specific about using `crane` here since I think we'd use that (or similar) tool to extract filesystem data from a container image, which feels like the more challenging part IMO. But as a dependency (or blocker?) for this issue we should probably develop the file domain first.

**Comment by brandtkeller** [comment]
%100 agree on a native-integration - in no way wanted to invalidate how valuable that may be. Figuring out which domains we will support is something I am hopeful we can navigate with demand signals and prototypes through the files domain.

**Comment by brandtkeller** [comment]
Look at discovery with #337 to prove out workflow
